packages/mask/src/plugins/CryptoartAI/SNSAdaptor/ActionBar.tsx

```diff
@@ -27,6 +28,11 @@ export function ActionBar(props: ActionBarProps) {
     const { classes } = useStyles()
     const { asset } = CollectibleState.useContainer()
 
+    const assetSource = useMemo(() => {
+        if (!asset.value || asset.error) return
+        return asset.value
+    }, [asset.value])
```

Comment:
why not extract to hook function?

Reply:
e.g:
```ts
function useAssetSource(asset) {
    return useMemo(() => !asset.value || asset.error ? null : asset.value, [asset.value])
}

function useAssetSourceFromContainer() {
    const { asset } = CollectibleState.useContainer()
    return useAssetSource(asset)
}

const assetSource = useAssetSourceFromContainer()
```

Reply:
No big difference,right? generally use `useMemo` to improve preofrmance